forge/src/main/java/whocraft/tardis_refined/registry/forge/ItemRegistryImpl.java

```diff
@@ -1,24 +1,41 @@
 package whocraft.tardis_refined.registry.forge;
 
+import net.minecraft.network.chat.Component;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.world.flag.FeatureFlagSet;
 import net.minecraft.world.item.CreativeModeTab;
+import net.minecraft.world.item.CreativeModeTabs;
+import net.minecraft.world.item.Item;
 import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.level.block.Block;
 import net.minecraft.world.level.block.Blocks;
+import net.minecraftforge.event.CreativeModeTabEvent;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+import net.minecraftforge.fml.common.Mod;
 import whocraft.tardis_refined.TardisRefined;
 import whocraft.tardis_refined.registry.BlockRegistry;
 import whocraft.tardis_refined.registry.ItemRegistry;
+import whocraft.tardis_refined.registry.RegistrySupplier;
 
+@Mod.EventBusSubscriber(modid = TardisRefined.MODID, bus = Mod.EventBusSubscriber.Bus.MOD)
 public class ItemRegistryImpl {
-
-    public static CreativeModeTab TAB = new CreativeModeTab(TardisRefined.MODID) {
-        @Override
-        public ItemStack makeIcon() {
-            return new ItemStack(BlockRegistry.GLOBAL_CONSOLE_BLOCK.get());
-        }
-    };
+    public static CreativeModeTab TAB;
 
     public static CreativeModeTab getCreativeTab() {
         return TAB;
     }
 
-
+    @SubscribeEvent
+    public static void addCreativeTab(CreativeModeTabEvent.Register event) {
+        event.registerCreativeModeTab(new ResourceLocation(TardisRefined.MODID, TardisRefined.MODID),
+                builder -> builder.icon(() -> new ItemStack(BlockRegistry.GLOBAL_CONSOLE_BLOCK.get()))
+                        .displayItems((featureFlagSet, output, bl) -> {
+                            for (RegistrySupplier<?> itemRegistrySupplier : ItemRegistry.FORGE_CREATIVE_ITEMS) {
+                                output.accept((Item) itemRegistrySupplier.get());
+                            }
+                            for (Item item : ItemRegistry.CREATIVE_ITEMS) {
+                                output.accept(item);
+                            }
```

Comment:
Looking good, but can we please retain the use of the CreativeModeTabEvent.BuildContents? We should try to stick to the convention defined by Forge, this event is likely added for a reason.

Reply:
Of course, I will get that done in a moment.